src/nih_wayfinding/app/scripts/routing/turnamenitys-service.js

```diff
@@ -0,0 +1,76 @@
+
+(function () {
+    'use strict';
+
+    /* ngInject */
+    function TurnAmenitys ($http) {
+
+        var overpassUrl = 'http://overpass-api.de/api/interpreter';
+
+        var module = {
+            get: get,
+            attach: attach
+        };
+
+        return module;
+
+        function overpassQuery(x, y) {
+            var meters = 75;
+            var q = '[out:json];' +
+                    'node(around:' + meters +
+                    ',' + y +
+                    ',' + x +
+                    ')' +
+                    '[amenity][name][amenity!="bicycle_rental"];' +
+                    'out;';
+            return q;
+        }
+
+        /**
+         * Look up an amenity from OpenStreetMap to add to the turn by turn
+         * directions. Returns a promise.
+         *
+         * @param x longitude
+         * @param y latitude
+         */
+        function get(x, y) {
+
+            return $http.get(overpassUrl, {
+                params: {
+                    'data': overpassQuery(x, y)
+                }
+               // headers: {
+               //     'Origin': 'http://localhost:8001/'
+               // }
+            }).then(function (data) {
+                if(data.data.elements.length === 0) {
+                    return null;
+                }
+                return _.first(data.data.elements);
+            });
+        }
+
+        /**
+         * Asynchronously attach OSM amenity to LineString based on the
```

Comment:
Oh, you're totally right. I somehow missed the get() call in there.

This does feel a bit anti-patterny, mostly because you have no way of knowing when the operation is finished. 

It might make more sense to have this service simply have a method that takes a geojson feature, makes the request and returns a promise that resolves when the new amenity feature is attached. You could still call that method for each linestring feature, or more defensively request amenities as the user approaches the next line string in the navigation view.

Ideas @flibbertigibbet ?